app/assets/javascripts/discourse/app/components/user-menu/menu.js

```diff
@@ -210,7 +211,9 @@ export default class UserMenu extends Component {
 
   @bind
   onRouteChange() {
-    this.args.closeUserMenu();
+    if (this.header.userVisible) {
+      this.args.closeUserMenu();
+    }
```

Comment:
alternatively we could do:

```gjs
// header.gjs
toggleUserMenu(value) {
  this.header.userVisible = value ?? !this.header.userVisible;
```

```hbs
// header/user-menu-wrapper.gjs
<UserMenu @closeUserMenu={{fn @toggleUserMenu false}} />
```

…since calling `closeUserMenu` should never open that menu 😄

Reply:
@CvX okay yes that makes sense, will change :+1:

Reply:
@CvX any inkling though on why this particular "new feature" notification gets the `routeDidChange` event and none others do?

Reply:
I debugged it for a bit and my best guess is that it's a timing issue. Other routes (topic, badges) do way more async stuff and that's why closing the menu can resolve before the end transition:

<img width="550" alt="image" src="https://github.com/user-attachments/assets/92adbc40-a8b0-4999-931a-be857382f7a9" />

while the whats-new route finishes its transition in one go:

<img width="550" alt="image" src="https://github.com/user-attachments/assets/b3879ced-aaee-4223-a7ae-597de2208762" />